components/brave_referrals/browser/brave_referrals_service.cc

```diff
@@ -125,9 +132,47 @@ void DeletePromoCodeFile(const base::FilePath& promo_code_file) {
   }
 }
 
+#if BUILDFLAG(IS_MAC)
+std::string ReadPromoCodeFromXattr() {
+  static constexpr char kRefCodeAttr[] = "com.brave.refcode";
+  base::FilePath bundle_path = base::apple::OuterBundlePath();
+  std::vector<char> value;
+  int result_errno =
+      brave::GetFileExtendedAttribute(bundle_path, kRefCodeAttr, &value);
+  if (result_errno == ENOATTR) {
+    LOG(WARNING) << "Could not get promo code from " << bundle_path
```

Comment:
VLOG

Reply:
Fixed in https://github.com/brave/brave-core/pull/20262/commits/83030888a121b765276ee249336f92af0561e05f